options/components/webTopContent/top-contents/WebKeywordSearchBox.jsx

```diff
@@ -1,11 +1,49 @@
-import { faMagnifyingGlass, faRotate } from "@fortawesome/free-solid-svg-icons";
+import { faMagnifyingGlass } from "@fortawesome/free-solid-svg-icons";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { useContext } from "react";
 
 import { WebSearchContext } from "../../../context/WebSearchContext";
 
 export default function WebKeywordSearchBox() {
-  const { searchValue } = useContext(WebSearchContext);
+  const { reSearchKeyword, setReSearchKeyword, filteredData, setFilteredData } =
+    useContext(WebSearchContext);
+
+  const handleChangeReSearchKeyword = (event) => {
+    setReSearchKeyword(event.target.value);
+  };
+
+  const handleReSearchResults = () => {
+    const copiedData = JSON.parse(JSON.stringify(filteredData.data));
+    const historyArray = [];
+    const seenUrls = new Set();
+
+    copiedData.forEach((innerData) => {
+      Object.values(innerData).forEach((data) => {
+        data.urlAllText.forEach((objectItem) => {
+          if (objectItem.includes(reSearchKeyword)) {
+            if (!seenUrls.has(data.url)) {
+              seenUrls.add(data.url);
+              historyArray.push({ [data.url]: data });
+            }
+          }
+        });
+      });
+    });
```

Comment:
```suggestion
    copiedData.forEach((innerData) => {
      Object.values(innerData).forEach((data) => {
        const match = data.urlAllText.some((text) => text.includes(reSearchKeyword));
        if (match && !seenUrls.has(data.url)) {
          seenUrls.add(data.url);
          historyArray.push({ [data.url]: data });
        }
      });
    });
```
forEach가 중첩되어 있는 형태에서 가독성을 더 개선할 방법이 있을 것 같습니다.
또한, forEach는 break가 되지 않습니다. 따라서,
마지막 forEach의 경우 해당 urlAllText가 검색어를 포함하고 있는지 여부를 검사하는 forEach문으로 보입니다.
이런 forEach는 some()으로 변경해서 포함여부가 확인되면 배열을 전부 순회하지 않고 빠르게 리턴할 수 있을 것 같습니다.
여전히 시간복잡도는 O(n^3)이지만 some으로 바꾸게 되면 조금의 성능향상도 노릴 수 있을 것 같습니다.

Reply:
data를 순회해서 꺼내야한다는생각에 forEach를 고집하게 되었었습니다.
그러나 수보님께서 말씀하신대로 하나만 검색어가 포함되면 검색결과 대상이기 때문에 본문을 전부 순회할 필요가 없는 부분 체크할 수 있었습니다.
더불어 some함수를 알고 만 있었는데 용도에 맞게 떠올리지 못하였는데 덕분에 사용해 볼 수 있었습니다.
시간복잡도가 높은 와중 성능개선을 위한 피드백 감사합니다.